### PR TITLE
jobs: add some more debugging to TestJobControlByType

### PR DESCRIPTION
**Pull request by knz**
This test is currently flaky (#99200, #98558). However, its failures do not tell us much.

This PR adds some additional debugging.

Epic: None

### CODE REVIEW

**Review by adityamaru — APPROVED** [review]
This will help a lot, thanks!

**Comment by adityamaru** [comment]
I can look into the test failures in CI now that we have this improved output.

File: pkg/jobs/delegate_control_test.go [hunk 1]
```diff
@@ -425,32 +455,38 @@ func TestJobControlByType(t *testing.T) {
...
 					commandQuery,
 				)
-				require.NoError(t, err)
+				require.NoError(t, err, "%s", delayedShowJobs(t, subJobs))
 
-				// Jobs in the starting state should be affected
+				// Jobs in the starting state should be affected.
 				numExpectedJobsAffected := numJobsPerStatus * len(tc.startingStates)
-				require.Equal(t, numExpectedJobsAffected, numEffected)
+				require.Equal(t, numExpectedJobsAffected, numEffected+1, "%s", delayedShowJobs(t, subJobs))
```
Comment (adityamaru) [inline pkg/jobs/delegate_control_test.go:482]: was this +1 to force a failure and test the additional debugging?

Comment (knz) [reply pkg/jobs/delegate_control_test.go:482]: yes, sorry it was indeed to test the additional debugging. I'll remove it before this can merge.

**Comment by knz** [comment]
bors r=adityamaru

**Comment by knz** [comment]
blathers backport 23.1

**Pull request merged**